scylla/src/transport/cluster.rs

```diff
@@ -84,28 +84,37 @@ impl Cluster {
         pool_config: PoolConfig,
         fetch_schema_metadata: bool,
     ) -> Result<Cluster, QueryError> {
-        let cluster_data = Arc::new(ArcSwap::from(Arc::new(ClusterData {
-            known_peers: HashMap::new(),
-            ring: BTreeMap::new(),
-            keyspaces: HashMap::new(),
-            all_nodes: Vec::new(),
-            datacenters: HashMap::new(),
-        })));
-
         let (refresh_sender, refresh_receiver) = tokio::sync::mpsc::channel(32);
         let (use_keyspace_sender, use_keyspace_receiver) = tokio::sync::mpsc::channel(32);
         let (server_events_sender, server_events_receiver) = tokio::sync::mpsc::channel(32);
 
+        let mut metadata_reader = MetadataReader::new(
+            initial_peers,
+            pool_config.connection_config.clone(),
+            pool_config.keepalive_interval,
+            server_events_sender,
+            fetch_schema_metadata,
+        );
+
+        let metadata = match metadata_reader.read_metadata().await {
+            Ok(data) => data,
+            Err(err) => {
+                warn!(
+                    error = ?err,
+                    "Initial metadata read failed, proceeding with dummy metadata",
```

Comment:
let's be specific, this is a user-facing error: "proceeding with metadata consisting only of the initial peers list"

Reply:
(or alike, feel free to rephrase)

Reply:
Done, I also mentioned how the failed metadata fetch might affect the user.